File: Go/LinkedList.go
package main
import "fmt"

type Node struct {
	data int
	link *Node
}

var head *Node

func addNode(data int){
	if head == nil {
		head = &Node{data, (&Node{})}
	} else {
		temp := &Node{data, head}
		head = temp
	}
}

func deleteNode(data int){
	temp := head
	prev_node := head
	for {
		if temp == nil {
			break
		}
		if temp.data == data {
			prev_node.link = temp.link
		}
		prev_node = temp
		temp = temp.link
	}
}

func findNode(data int) bool {
	temp := head
	var flag bool = false
	for {
		if temp == nil {
			break
		}
		if temp.data == data {
			flag = true
		}
		temp = temp.link
	}
	return flag
}

func size() int {
	temp := head
	var counter int = 0
	for {
		if temp == nil {
			break
		}
		counter++
		temp = temp.link
	}
	return counter
}

func printList(){
	temp := head
	for {
		if temp == nil {
			break
		}
		fmt.Println(temp.data)
		temp = temp.link
	}
}

func main(){
	addNode(1)
	addNode(3)
	addNode(5)
	addNode(7)
	addNode(9)

	printList()

	fmt.Println(size())
	fmt.Println(findNode(5))
	deleteNode(5)
	fmt.Println(findNode(5))

	printList()
}

File: PHP/LinkedList.php
<?php
class Node {
    function __contruct($data, $node){
        $this->data = $data;
        $this->link = $node;
    }

    function setLink($node){
        $this->link = $node;
    }

    function getLink(){
        return $this->link;
    }
}

class LinkedList {
    function __contruct(){
        $this->head = null;
    }

    function addNode($data){
        $temp = new Node($data, null);
        if ($this->head === null){
            $this->head = $temp;
        } else {
            $temp->link = $this->head;
            $this->head = $temp;
        }
    }

    function deleteNode($data){
        $temp = $this->head;
        $prevNode = null;
        while($temp !== null){
            if ($temp->data === data){
                $prevNode->link = $temp->link;
            }
            $prevNode = $temp;
            $temp = $temp->link;
        }
    }

    function findNode($data){
        $temp = $this->head;
        $flag = false;
        while($temp !== null){
            if ($temp->data === data){
                $flag = true;
                break;
            }
            $temp = $temp->link;
        }
        return $flag;
    }

    function size(){
        $temp = $this.head;
        $counter = 0;
        while($temp !== null){
            $counter++;
            $temp = $temp->link;
        }
        return $counter;
    }

    function printList(){
        $temp = $this.head;
        while($temp !== null){
            echo $temp->data;
            $temp = $temp->link;
        }
    }
}

$linkedLinked = new LinkedList();

$linkedLinked->addNode(1);
$linkedLinked->addNode(3);
$linkedLinked->addNode(5);
$linkedLinked->addNode(7);
$linkedLinked->addNode(9);

$linkedLinked->printlList();
echo $linkedLinked->size();

echo $linkedLinked->findNode(5);
$linkedLinked->deleteNode(5);
echo $linkedLinked->findNode(5);
$linkedLinked->printList();

?>
